server/app/services/question/QuestionServiceImpl.java

```diff
@@ -126,6 +136,12 @@ public void archiveQuestion(Long id) throws InvalidUpdateException {
     if (question.isEmpty()) {
       throw new InvalidUpdateException("Did not find question.");
     }
+    ActiveAndDraftQuestions activeQuestions =
+        readOnlyQuestionService().getActiveAndDraftQuestions();
+    if (activeQuestions.getDeletionStatus(question.get().getQuestionDefinition().getName())
+        != DeletionStatus.DELETABLE) {
```

Comment:
suggestion (blocking): use .equals for enums

Reply:
Done.